index.js

```diff
@@ -49,5 +50,30 @@ module.exports = {
   format: function(text, opts) {
     return formatWithShebang(text, normalizeOptions(opts));
   },
-  version: version
+  version: version,
+
+  __debug: {
+    // Doesn't handle shebang for now
+
+    formatDoc: function(doc, opts) {
+      opts = normalizeOptions(opts);
+      const debug = printDocToDebug(doc);
+      const str = format(debug, opts);
+      return str;
+    },
+
+    printToDoc: function(text, opts) {
+      opts = normalizeOptions(opts);
+      const ast = parse(text, opts);
+      attachComments(text, ast, opts);
+      const doc = printAstToDoc(ast, opts);
+      return doc;
+    },
+
+    printDocToString: function(doc, opts) {
```

Comment:
I'm not yet using this. What I want to do is to let you edit the doc editor and generate the printed version from it. This will come in handy :)